packages/auth/src/client/service.ts

```diff
@@ -166,14 +276,126 @@ function verifyClientDisplay(
   )
 }
 
-function verifyJwk(jwk: JWKWithRequired, keys: RegistryKey): boolean {
-  // TODO: update this to reflect eventual shape of response from registry
+function isJwkViable(keys: RegistryKey): boolean {
   return !!(
-    !keys.revoked &&
-    keys.exp &&
-    new Date() < new Date(keys.exp * 1000) &&
-    keys.nbf &&
-    new Date() >= new Date(keys.nbf * 1000) &&
-    keys.x === jwk.x
+    (!keys.exp || new Date() < new Date(keys.exp * 1000)) &&
+    (!keys.nbf || new Date() >= new Date(keys.nbf * 1000))
   )
 }
+
+function verifyJwk(jwk: JWKWithRequired, keys: RegistryKey): boolean {
+  // TODO: update this to reflect eventual shape of response from registry
+  return !!(!keys.revoked && isJwkViable(keys) && keys.x === jwk.x)
+}
+
+async function tokenHttpsigMiddleware(
```

Comment:
Musing here.... This middleware seems to do many thing that are path dependent.. Does it make sense for it to be a singular middleware?